src/main/java/com/softserveinc/ita/pageobjects/modals/BaseModal.java

```diff
@@ -0,0 +1,42 @@
+package com.softserveinc.ita.pageobjects.modals;
+
+import io.qameta.allure.Step;
+
+import static com.codeborne.selenide.Condition.*;
+import static com.codeborne.selenide.Selenide.$x;
+import static com.softserveinc.ita.pageobjects.util.DataProvider.INPUT_TEMPLATE;
+import static com.softserveinc.ita.pageobjects.util.DataProvider.TEXTAREA_TEMPLATE;
+import static java.lang.String.format;
+import static java.time.Duration.*;
+
+public class BaseModal<T extends BaseModal<T>> {
+    @Step("Adding form modal: Set {formControlName} title")
+    public T setValueFor(String formControlName, String value) {
+        $x(format(INPUT_TEMPLATE, formControlName))
+                .should(appear, ofSeconds(5))
+                .sendKeys(value);
+
+        return (T) this;
+    }
+
+    @Step("Adding form modal: Set {formControlName} description")
+    public T setDescriptionFor(String formControlName, String description) {
+        $x(format(TEXTAREA_TEMPLATE, formControlName))
+                .should(appear, ofSeconds(5))
+                .sendKeys(description);
+
+        return (T) this;
+    }
+
+    public boolean isAddButtonEnabled() {
+        return $x("//button[@type='submit']")
```

Comment:
also, you can use this locator for adding a new item (faculty, specialty, group), so you could create a common method here for this, instead of these methods ("addNewSubject", "addNewSpecialty")